src/Slidecoding/Types.hs

```diff
@@ -50,10 +56,50 @@ data Context = Context { name            :: Name         -- A name for debugging
                        }
 
 data Presentation = Presentation { rootDir :: FilePath
+                                 , distDir :: FilePath
                                  , meta    :: Metadata
                                  } deriving Show
 
-data Metadata = Metadata { title :: Text
+data Metadata = Metadata { title  :: String
+                         , design :: Design
                          } deriving Show
 
+type Pixel = Int
+data Design = Design { width  :: Pixel
+                     , height :: Pixel
+                     , theme  :: Theme
+                     , icon   :: Maybe FilePath
+                     } deriving Show
+data Theme = Web20 | Neon | Swiss | Custom FilePath | Patch Theme FilePath deriving Show
```

Comment:
`Web20 | Neon | Swiss` curious to have such constants in this module. Those values are specific to deck.js and shouldn't be referenced there. Replace those 3 constructors with `Theme ThemeName` and let the deck.js adapter know what to do.